.github/workflows/release-e2e.yaml

```diff
@@ -0,0 +1,57 @@
+---
+name: release-e2e
+
+on:
+  release:
+    types: [created]
+  workflow_dispatch:
+
+jobs:
+  release-e2e-kind:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '~1.18'
+
+    - name: install_kind_binary
+      env:
+        KIND_VERSION: v0.12.0
+        KIND_OS: kind-linux-amd64
+      run: |
+        curl -sLo kind "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/${KIND_OS}"
+        chmod +x kind
+        sudo mv kind /bin/
```

Comment:
should we `make kind` and then use `hack/tools/bin/kind`? That way we get the version of kind that the repo specifies?

Right now, we're pretty loose (read: we don't care what it is) with the kind version in the Makefile. Perhaps we should?

Reply:
I don't see a `kind` target in the rukpak Makefile.

Reply:
Yes, I think it makes sense to pin a specific version of kind in our Makefile, to make our CI more hermetic

Reply:
Would it make sense to pin the Makefile to the most recent release? Set the version in the Makefile directly and manually update it? Something else entirely?

Reply:
This seems like a reasonable request so I opened https://github.com/operator-framework/rukpak/issues/365 which can be done out of band from these changes.

Reply:
Thanks, Tim. Assigned that to myself. Might as well do this correctly.

Reply:
Merged in the change - will rebase and use it in this PR.